contracts/core/OriginCore.sol

```diff
@@ -599,4 +658,83 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
         return transitionObject.kernelHash == _kernelHash;
     }
 
+    /**
+     * @notice Function to commit meta-block if 2/3rd majority is achieved.
+     *
+     * @param _kernelHash The hash of the current kernel.
+     * @param _transitionHash The hash of the transition object which is proposed
+     *                        with meta-block.
+     * @param _height Height of current meta-block.
+     * @param _verifiedWeight Total verified weight.
+     */
+    function commitMetaBlock(
+        bytes32  _kernelHash,
+        bytes32 _transitionHash,
+        uint256 _height,
+        uint256 _verifiedWeight
+    )
+        private
+    {
+
+        uint256 requiredWeight = MetaBlock.requiredWeightForSuperMajority(
+            stake.totalWeightAtHeight(height)
+        );
+
+        if (_verifiedWeight < requiredWeight) {
+            return;
+        }
```

Comment:
Shouldn't an `assert(_verified >= requiredWeight)` be sufficient here? Should have been verified before, right? Or maybe even skip the check here? Gas vs. correctness 🧐

Reply:
I have moved the check before method call. 
However, we don't want to revert transaction here. It's a conditional check instead of `assert`.